### PR TITLE
Build Object-Store

### PR DESCRIPTION
**Pull request by beeender**
**DO NOT SQUASH WHEN MERGING!! #2915 SHOULD BE MERGED FIRST!!!**

All credits go to @kneth 

1. Fixed compile errors when build with NDK, and changes are intended to be in one separated commit which will be pushed to os repo as well.
2. Continue kg's work to build the object store source.

### CODE REVIEW

**Comment by beeender** [comment]
CI fails because of https://code.google.com/p/android/issues/detail?id=62648
It fails on arm ARCH only because `exception_ptr` is not available .... :(

File: realm/realm-jni/build.gradle [hunk 1]
```diff
@@ -75,7 +75,8 @@ def toolchains = [
 ]
 
 def allTargets = [
-    new Target( name:'arm',     abi:'armeabi',     toolchain:toolchains.find {it.name == 'arm'},    cflags:[ '-mthumb' ] ),
+    // gcc for armeabi has problem to compile std::excpetion_ptr. Disable this ABI for now. We might drop it or use the new NDK clang to compile it.
+    //new Target( name:'arm',     abi:'armeabi',     toolchain:toolchains.find {it.name == 'arm'},    cflags:[ '-mthumb' ] ),
```
Comment (stk1m1) [inline realm/realm-jni/build.gradle]: Are we going to stop `armabi` support for sure?

Comment (kneth) [inline realm/realm-jni/build.gradle]: clang seems to support `exception_ptr` but we are still investigating.

Comment (stk1m1) [inline realm/realm-jni/build.gradle]: > clang seems to support `exception_ptr` but we are still investigating.

Never mind about my question. I almost forgot the point. Thanks for the reminder. 👍

**Pull request merged**